cnf-certification-test/lifecycle/suite.go

```diff
@@ -584,23 +581,30 @@ func TestPodTolerationBypass(env *provider.TestEnvironment) {
 	testhelper.AddTestResultLog("Non-compliant", podsWithRestrictedTolerationsNotDefault, tnf.ClaimFilePrintf, ginkgo.Fail)
 }
 func testStorageRequiredPods(env *provider.TestEnvironment) {
-	oc := clientsholder.GetClientsHolder()
-	var nonCompliantPods []*provider.Pod
-	for _, pod := range env.Pods {
-		for _, podOwner := range pod.ObjectMeta.GetOwnerReferences() {
-			if podOwner.Kind != statefulSet {
-				continue
-			}
-			statefulset, err := oc.K8sClient.AppsV1().StatefulSets(pod.Namespace).Get(context.TODO(), podOwner.Name, apiv1.GetOptions{})
-			if err != nil {
-				tnf.ClaimFilePrintf(err.Error())
+	var podsWithLocalStorage []string
+	for _, put := range env.Pods {
+		for pvIndex := range put.Spec.Volumes {
+			// Skip any nil persistentClaims.
+			if put.Spec.Volumes[pvIndex].PersistentVolumeClaim == nil {
 				continue
 			}
-			if statefulset.Spec.ServiceName == localStorage {
-				nonCompliantPods = append(nonCompliantPods, pod)
-				continue
+			// We have the list of pods/volumes/claims.
+			// Look through the storageClass list for a match.
+			for _, s := range env.StorageList {
+				if s.Name == put.Spec.Volumes[pvIndex].PersistentVolumeClaim.ClaimName  {
+					// At this point we have found the storage class attached to the pod.
+					// We need to check if this storageClass is no-provisioner.
+					if s.Provisioner == "kubernetes.io/no-provisioner" {
+						tnf.ClaimFilePrintf("This is a bad pod: The pod using in local storage")
+						podsWithLocalStorage = append(podsWithLocalStorage, put.String())
+						// Break out of the storageList loop.
+						continue	
```

Comment:
Looking at this again, I think we should keep a "localStorageFound" boolean per pod and then just only do `podsWithLocalStorage = append(podsWithLocalStorage, put.String())` once per pod.

Reply:
We can save map that we have details on the pod and also the boolean variable "localStorageFound" and we save it global and then we have information about every pod.
and if we doing like this so, i think that we don't need the message in line 598.

What do you think? @sebrandon1

Reply:
I think the `continue` with the `boolean` approach will have less iterations when looping instead of just potentially adding the same pod to the map when/if the first volume fails.

We should keep the message for context for the user.

Reply:
> You should also add a bad test case yaml example to test both passing and failing pods in the partner repo /examples directory

I will do it in another PR (on repo partner)

Reply:
The changes have been uploaded @sebrandon1 ֿ